backend/main/models/community_users.go

```diff
@@ -98,29 +100,54 @@ func GetUsersForCommunityByType(db *s.Database, communityId, start, count int, u
 	return users, totalUsers, nil
 }
 
-func GetCommunityLeaderboard(db *s.Database, communityId, start, count int) ([]LeaderboardUser, int, error) {
-	var users = []LeaderboardUser{}
-	err := pgxscan.Select(db.Context, db.Conn, &users,
+func GetCommunityLeaderboard(db *s.Database, communityId, start, count int) ([]LeaderboardUserPayload, int, error) {
+	var userAchievements = [] struct {
+		Address   string
+		NumVotes  int
+		EarlyVote int
+	}{}
+	var leaderboardUsers = []LeaderboardUserPayload{}
+	var defaultEarlyVoteWeight = 1
+
+	sql := fmt.Sprintf(
 		`
-		SELECT v.addr, count(*) AS score FROM votes v 
-		JOIN proposals p ON p.id = v.proposal_id
-		WHERE p.community_id = $1
-		GROUP BY v.addr
-		ORDER BY score DESC
-		LIMIT $2 OFFSET $3
-		`, communityId, count, start)
+		SELECT v.addr as address, count(*) as num_votes, 
+			CASE WHEN a.early_vote is NULL THEN 0 ELSE a.early_vote END as early_vote 
+			FROM votes v 
+			LEFT OUTER JOIN proposals p ON p.id = v.proposal_id
+			LEFT OUTER JOIN (
+				SELECT * FROM crosstab(
+					$$SELECT addr, achievement_type, count(*) FROM community_users_achievements 
+					WHERE community_id = %d
+					GROUP BY addr, achievement_type
+					ORDER BY 1,2$$
+				) AS ct(address varchar(18), early_vote bigint)
+			) a ON v.addr = a.address
+			WHERE p.community_id = $1
+			GROUP BY v.addr, a.early_vote
+			LIMIT $2 OFFSET $3
+		`, communityId)
+
+	err := pgxscan.Select(db.Context, db.Conn, &userAchievements, sql, communityId, count, start)
 
 	if err != nil && err.Error() != pgx.ErrNoRows.Error() {
 		return nil, 0, err
 	} else if err != nil && err.Error() == pgx.ErrNoRows.Error() {
-		return []LeaderboardUser{}, 0, nil
+		return []LeaderboardUserPayload{}, 0, nil
+	}
+
+	for _, user := range userAchievements {
+		var leaderboardUser = LeaderboardUserPayload{}
+		leaderboardUser.Addr = user.Address
+		leaderboardUser.Score = user.NumVotes + (user.EarlyVote * defaultEarlyVoteWeight)
```

Comment:
Is the idea here that when we enable user editable weights that this value will not be 1? If so maybe it's better to add `earlyVoteWeight` as an argument to the method and assign a default value if it is not passed?

Reply:
yeah.. Go doesn't support optional fields, Golang enums can be useful if you need to pass different types into the same func tho!

For default values, you could use the constructor pattern, so where leaderboardUserPayload would have a function to initialise it with a default value.

Reply:
Yea, once get it all working, we can revisit when we bring in the custom settings support.